packages/mantine/src/components/table/table-edit-columns-visibility/TableEditColumnsVisibility.tsx

```diff
@@ -0,0 +1,91 @@
+    const adjustedMaxSelectableColumns = maxSelectableColumns
+        ? Math.min(maxSelectableColumns, filteredColumns.length)
+        : filteredColumns.length;
+
+    const remainingColumnsCount = adjustedMaxSelectableColumns
+        ? adjustedMaxSelectableColumns - selectedColumnsCount
+        : filteredColumns.filter((column) => !column.getIsVisible() && !nonHideableColumns?.includes(column.id)).length;
+
+    let footerText;
+    if (selectedColumnsCount === filteredColumns.length) {
+        footerText = 'All available columns selected';
+    } else if (adjustedMaxSelectableColumns && selectedColumnsCount >= adjustedMaxSelectableColumns) {
+        footerText = 'Maximum columns selected';
+    } else {
+        footerText = `You can choose ${remainingColumnsCount} more`;
+    }
+
+    if (filteredColumns.length <= 0) {
+        return null;
+    }
+
+    return (
+        <Grid.Col span="content" order={TableComponentsOrder.EditColumnsVisibility} py="sm" className={classes.root}>
+            <Popover position="bottom" shadow="md">
+                <Popover.Target>
+                    <Button variant={buttonVariant}>{`${label}${
+                        showVisibleCountLabel ? ` (${selectedColumnsCount})` : ''
+                    }`}</Button>
+                </Popover.Target>
+                <Popover.Dropdown miw={240}>
+                    <ScrollArea.Autosize mah={154}>
+                        <Stack>
+                            {filteredColumns
+                                .sort((a, b) =>
+                                    // sort to render nonHideableColumns first
+                                    nonHideableColumns?.includes(a.id) && !nonHideableColumns?.includes(b.id) ? -1 : 1,
+                                )
+                                .map((column) => (
+                                    <Checkbox
+                                        key={column.id}
+                                        label={columnNames?.[column.id] || column.id}
+                                        name={column.id}
+                                        checked={nonHideableColumns?.includes(column.id) ? true : column.getIsVisible()}
+                                        disabled={
+                                            nonHideableColumns?.includes(column.id) ||
+                                            (selectedColumnsCount >= adjustedMaxSelectableColumns &&
+                                                !column.getIsVisible())
+                                        }
+                                        onChange={column.getToggleVisibilityHandler()}
+                                    />
+                                ))}
+                        </Stack>
+                    </ScrollArea.Autosize>
+                    {maxSelectableColumns && (
+                        <>
+                            <Divider mb="xs" mt="sm" />
+                            <div>{footerText}</div>
```

Comment:
You don't need the div

```suggestion
                            {footerText}
```